src/backend/gl/src/window/surfman.rs

```diff
@@ -95,75 +94,69 @@ impl Instance {
             .bind_surface_to_context(&mut context, surface)
             .expect("TODO");
 
-        device.make_context_current(&context).expect("TODO");
+        device
+            .make_context_current(&context)
+            .expect("TODO");
 
         // Create a surface with the given context
         Surface {
             renderbuffer: None,
             swapchain: None,
             context: Starc::new(RwLock::new(context)),
-            device: Starc::new(RwLock::new(device)),
+            device: self.device.clone(),
```

Comment:
Does the surface really need to get a hold of the device?

Reply:
Yes because the drop implementation needs a device handle in order to clean up the surface and the context.

Reply:
We now have `Instance::destroy_surface`, where this logic needs to be moved into. The Drop shouldn't even be there.

Reply:
Ah, OK. One second and I'll fix that.

Reply:
There you go.